### DISCUSSION TITLE
USB sound card & ALSA output plugin settings

### DISCUSSION THREAD

**Discussion by hmalissa** [cmus/cmus#1089] [General]
I've recently got a USB sound card, but I can't figure out how to configure dsp.alsa.device, mixer.alsa.channel, and mixer.alsa.device to use a particular sound card for playback. dsp.alsa.device is set to 'default', and music is being played to whichever sound card happens to be assigned to ALSA card #0. What are legitimate values for these parameters? How do I select the external sound card (card #1)?

# aplay -l
**** List of PLAYBACK Hardware Devices ****
card 0: PCH [HDA Intel PCH], device 0: ALC256 Analog [ALC256 Analog]
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 0: PCH [HDA Intel PCH], device 3: HDMI 0 [HDMI 0]
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 0: PCH [HDA Intel PCH], device 7: HDMI 1 [HDMI 1]
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 0: PCH [HDA Intel PCH], device 8: HDMI 2 [HDMI 2]
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 0: PCH [HDA Intel PCH], device 9: HDMI 3 [HDMI 3]
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 0: PCH [HDA Intel PCH], device 10: HDMI 4 [HDMI 4]
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 1: Schiit [I'm Fulla Schiit], device 0: USB Audio [USB Audio]
  Subdevices: 1/1
  Subdevice #0: subdevice #0

**Comment by nefthy** [comment]
try the following: 
`set dsp.alsa.device=plughw:Schiit`
or `set dsp.alsa.device=hw:Schiit`
or `set dsp.alsa.device=plughw:1`
or `set dsp.alsa.device=hw:1`

**Comment by nefthy** [comment]
alternatively you can also change the default card in the alsa config: https://www.alsa-project.org/wiki/Setting_the_default_device

**Comment by hmalissa** [comment]
set dsp.alsa.device=plughw:Schiit and set dsp.alsa.device=plughw:1 works, thanks a lot.
set dsp.alsa.device=hw:Schiit and set dsp.alsa.device=hw:1 gives me 'Error: opening audio device: Invalid argument'.
